src/rgen/Microsoft.Macios.Generator/Emitters/BindingSyntaxFactory.Dlfcn.cs

```diff
@@ -0,0 +1,447 @@
+		=> throw new NotImplementedException ();
+
+	public static CompilationUnitSyntax SetIntPtr (string libraryName, string symbol, IntPtr value)
+		=> throw new NotImplementedException ();
+
+	public static CompilationUnitSyntax GetCGRect (string libraryName, string symbol)
+		=> GetConstant ("GetCGRect", libraryName, symbol);
+
+	/// <summary>
+	/// Generates a call for "Dlfcn.GetCGSize (libraryName, fieldName);"];
+	/// </summary>
+	/// <param name="libraryName">The library from where the field will be loaded.</param>
+	/// <param name="fieldName">The field name.</param>
+	/// <returns>A compilation unit with the desired Dlfcn call.</returns>
+	public static CompilationUnitSyntax GetCGSize (string libraryName, string fieldName)
+		=> GetConstant ("GetCGSize", libraryName, fieldName);
+
+	public static CompilationUnitSyntax SetCGSize (string libraryName, string symbol, /*CGSize*/ string value)
+		=> throw new NotImplementedException ();
+
+	/// <summary>
+	/// Generates a call for "Dlfcn.GetDouble (libraryName, fieldName);"];
+	/// </summary>
+	/// <param name="libraryName">The library from where the field will be loaded.</param>
+	/// <param name="fieldName">The field name.</param>
+	/// <returns>A compilation unit with the desired Dlfcn call.</returns>
+	public static CompilationUnitSyntax GetDouble (string libraryName, string fieldName)
+		=> GetConstant ("GetDouble", libraryName, fieldName);
+
+	public static CompilationUnitSyntax SetDouble (string libraryName, string symbol, double value)
+		=> throw new NotImplementedException ();
+
+	/// <summary>
+	/// Generates a call for "Dlfcn.GetFloat (libraryName, fieldName);"];
+	/// </summary>
+	/// <param name="libraryName">The library from where the field will be loaded.</param>
+	/// <param name="fieldName">The field name.</param>
+	/// <returns>A compilation unit with the desired Dlfcn call.</returns>
+	public static CompilationUnitSyntax GetFloat (string libraryName, string fieldName)
+		=> GetConstant ("GetFloat", libraryName, fieldName);
+
+	public static CompilationUnitSyntax SetFloat (IntPtr handle, string symbol, float value)
+		=> throw new NotImplementedException ();
+
+	/// <summary>
+	/// Generates a call for "Dlfcn.GetSizeF (libraryName, fieldName);"];
+	/// </summary>
+	/// <param name="libraryName">The library from where the field will be loaded.</param>
+	/// <param name="fieldName">The field name.</param>
+	/// <returns>A compilation unit with the desired Dlfcn call.</returns>
+	public static CompilationUnitSyntax GetSizeF (string libraryName, string fieldName)
+		=> GetConstant ("GetSizeF", libraryName, fieldName);
+
+	public static CompilationUnitSyntax GetSizeF (IntPtr handle, string fieldName, float value)
+		=> throw new NotImplementedException ();
+
+	/// <summary>
+	/// Generates a call for "Runtime.GetNSObject;lt&Foundation.NSArray;gt& (Dlfcn.GetIndirect (libraryName, fieldName))!;"];
```

Comment:
I think the escape sequences are wrong, also NSArray?:

```suggestion
	/// Generates a call for "Runtime.GetNSObject&lt;nsobjectType&gt; (Dlfcn.GetIndirect (libraryName, fieldName))!;"];
```